packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -331,9 +331,149 @@ const Devices = WebexPlugin.extend({
 
   // Registration method members
 
-  refresh() {}, // Refreshes the device's registration.
-  register() {}, // Registers a device.
-  unregister() {}, // Unregisters the device.
+  /**
+   * Refresh the current registered device if able.
+   *
+   * @returns {Promise<void, Error>}
+   */
+  refresh() {
+    this.logger.info('devices: refreshing');
+
+    // Validate that the device can be registered.
+    return this.canRegister(true)
+      .then(() => {
+        // Validate if the device is not registered and register instead.
+        if (!this.registered) {
+          this.logger.info('devices: device not registered, registering');
+
+          return this.register();
+        }
+
+        // Merge body configurations, overriding defaults.
+        const body = {
+          ...(this.serialize()),
+          ...(this.config.body ? this.config.body : {})
+        };
+
+        // Remove unneeded properties from the body object.
+        delete body.features;
+        delete body.mediaCluster;
+
+        // Append a ttl value if the device is marked as ephemeral.
+        if (this.config.ephemeral) {
+          body.ttl = this.config.ephemeralDeviceTTL;
+        }
+
+        // Merge header configurations, overriding defaults.
+        const headers = {
+          ...(this.config.defaults.headers ? this.config.defaults.headers : {}),
+          ...(this.config.headers ? this.config.headers : {})
+        };
+
+        return this.request({
+          method: 'PUT',
+          uri: this.url,
+          body,
+          headers
+        })
+          .then((response) => this.processRegistrationSuccess(response))
+          .catch((reason) => {
+            // Handle a 404 error, which indicates that the device is no longer
+            // valid and needs to be registered as a new device.
+            if (reason.statusCode === 404) {
+              this.logger.info('devices: refresh failed, device is not valid');
+              this.logger.info('devices: attempting to register a new device');
+
+              this.clear();
+
+              return this.register();
+            }
+
+            return Promise.reject(reason);
+          });
+      })
+      .catch((error) => {
```

Comment:
Would this catch the promise rejection on line 392 for the 404 error? If so, the error message isn't correct.

Reply:
This can be fixed by simply removing the catch block to follow the `this.canRegister()` method. This will simplify the registration method's logic as well.